pytorch_lightning/utilities/parsing.py

```diff
@@ -115,11 +124,11 @@ def get_init_args(frame) -> dict:
     self_var, args_var, kwargs_var = parse_class_init_keys(cls)
     filtered_vars = [n for n in (self_var, args_var, kwargs_var) if n]
     exclude_argnames = (*filtered_vars, '__class__', 'frame', 'frame_args')
-
     # only collect variables that appear in the signature
     local_args = {k: local_vars[k] for k in init_parameters.keys()}
     local_args.update(local_args.get(kwargs_var, {}))
     local_args = {k: v for k, v in local_args.items() if k not in exclude_argnames}
+    local_args = apply_to_collection(local_args, DictConfig, resolve_dict_config)
```

Comment:
does this convert everything to Hydra even you pass pure dict?

Reply:
It will be triggered only on DictConfig.

Reply:
so you say whenever you have Hydra installed, you will convert any input to `DictConfig`, correct?

Reply:
No, it will convert a DictConfig with interpolation to a resolved one.